tools/bin/cdk-auto-release/update-changelog.sh

```diff
@@ -0,0 +1,413 @@
+        error_exit "Failed to fetch from origin master."
+    fi
+}
+
+# Function to find the merge base between HEAD and origin/master
+# Usage: find_merge_base
+# Returns: The merge base commit hash or exits on error
+find_merge_base() {
+    local merge_base
+    log_info "Finding merge base between HEAD and origin/master..."
+    merge_base=$(git merge-base HEAD origin/master)
+
+    if [ -z "$merge_base" ]; then
+      error_exit "Could not find a common ancestor between HEAD and origin/master."
+    fi
+
+    log_info "Merge base commit: $merge_base"
+    echo "$merge_base" # Return merge base commit hash
+}
+
+# Function to find modified connector folders since the merge base
+# Usage: find_modified_connectors <merge_base_commit>
+# Returns: A string containing unique connector folder names (e.g., "source-postgres\ndestination-bigquery")
+find_modified_connectors() {
+    local merge_base_commit="$1"
+    local modified_files connector_folders
+
+    log_info "Checking for modified connector files since merge base ${merge_base_commit}..." >&2
+    modified_files=$(git diff --name-only --diff-filter=M "$merge_base_commit" HEAD -- 'airbyte-integrations/connectors/*') # Be specific
+
+    if [ -z "$modified_files" ]; then
+      log_info "No connector files were modified on the current branch compared to its merge-base with origin/master." >&2
+      echo "" # Return empty string
+    else
+      log_info "Modified connector files found: $modified_files" >&2
+      # Extract unique connector folder names (e.g., source-postgres)
+      connector_folders=$(echo "$modified_files" | \
+          # --- Old Line (remove or comment out) ---
+          # grep -oP 'airbyte-integrations/connectors/\K[^/]+' | \
+          # --- New Line using awk ---
+          awk -F'/' '$1=="airbyte-integrations" && $2=="connectors" && $3!="" { print $3 }' | \
+          sort -u) # Use sort -u for unique sort
+      log_info "Unique modified connector folders: $connector_folders" >&2
+      echo "$connector_folders" # Return the list
+    fi
+}
+
+# Function to get the PR number associated with a given branch
+# Usage: get_pr_number "feature/branch-name"
+# Returns: The PR number or exits if not found or 'gh' fails
+get_pr_number() {
+    local branch_name="$1"
+    local pr_num
+
+    log_info "Attempting to find PR number for branch: ${branch_name}"
+    # Ensure gh is installed
+    if ! command -v gh &> /dev/null; then
+        error_exit "'gh' command not found. Please install the GitHub CLI."
+    fi
+
+    # Use gh pr list to find the PR number for the specified head branch
+    pr_num=$(gh pr list --state open --head "$branch_name" --json number --jq '.[0].number // empty')
+
+    if [ -z "$pr_num" ]; then
+        error_exit "Could not find an open PR number for branch '${branch_name}'. Make sure the PR exists and the branch name is correct."
+    fi
+
+    log_info "Found PR number: ${pr_num}"
+    echo "$pr_num" # Return the PR number
+}
```

Comment:
I think this can be replaced by the GH context: https://stackoverflow.com/questions/59077079/how-to-get-pull-request-number-within-github-actions-workflow and passed into the workflow to simplify things.

Reply:
Will update and test

Reply:
I tested it and the issue is that we don't send it from a PR, we are triggering a workflow directly so there is no PR number.

Reply:
Painful. so because this is triggered by a slash command, it doesn't have the github context populated, even though the command is triggered from a PR?

Reply:
It might not be, I've been testing it using the workflow dispatch.

Reply:
I was thinking that it is up to the user. Either through a workflow dispatch or a slash command. Which makes me think that I forget to add it to the slash command dispatch.

Reply:
For example for my tests during the development, I used the gh command line to trigger the workflow.